### PR TITLE
Scott Nicholes project 02 draw triangle

### PR DESCRIPTION
**Pull request by scott56gps**
trianglescreenshot

The main thing I learned from this exercise was to make sure I am comparing the right vertices.  I had to go back and double check that I was using the correct maxs, mins and such.

### CODE REVIEW

File: pipeline.cpp [hunk 1]
```diff
@@ -53,6 +53,19 @@ void processUserInputs(bool & running)
...
+/****************************************
+ * COMPUTE_DETERMINANT
+ * Computes the determinant for a set of
+ * 2 Verticies, given in Matrix Form
+ * thus:
+ * 
+ * [a b]
+ * [c d]
+ * *************************************/
+double computeDeterminant(double a, double b, double c, double d) {
```
Comment (Brooken2) [inline pipeline.cpp]: This looks good!

File: pipeline.cpp [hunk 2]
```diff
@@ -78,9 +91,51 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
  * Renders a triangle to the target buffer. Essential 
  * building block for most of drawing.
  ************************************************************/
-void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
+void DrawTriangle(Buffer2D<PIXEL> &target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
     // Your code goes here
+    // target[(int)triangle[0].y][(int)triangle[0].x] = attrs[0].color;
+
+    int maxX = MAX3(triangle[0].x, triangle[1].x, triangle[2].x);
```
Comment (Brooken2) [inline pipeline.cpp]: MAX3 is a cool thing to know about!

File: pipeline.cpp [hunk 3]
```diff
@@ -180,13 +235,14 @@ int main()
...
+        processUserInputs(running);
 
         // Refresh Screen
-        //clearScreen(frame);
+        clearScreen(frame);
 
-        TestDrawPixel(frame);
+        //TestDrawPixel(frame);
         //GameOfLife(frame);
+        TestDrawTriangle(frame);
```
Comment (Justin9four) [inline pipeline.cpp]: Instead of just commenting it out I would remove these lines for elegance

File: pipeline.cpp [hunk 4]
```diff
@@ -78,9 +91,51 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
...
+        for (int y = minY; y <= maxY; y++) {
+            // Make a Vector that spans from the triangle[0] Vertex to the current values of x and y
+            Vertex testVector = {
+                x - triangle[0].x,
+                y - triangle[0].y,
+                1,
+                1
+            };
+
+            double det1 = computeDeterminant(testVector.x, side2.x, testVector.y, side2.y) / computeDeterminant(side1.x, side2.x, side1.y, side2.y);
```
Comment (Justin9four) [inline pipeline.cpp]: For elegance and readability, I would cut the line in half. It's too long.

**Pull request merged**